returnDetail.go

```diff
@@ -265,7 +265,13 @@ func (rd *ReturnDetail) String() string {
 
 // Validate performs image cash letter format rule checks on the record and returns an error if not Validated
 // The first error encountered is returned and stops the parsing.
-func (rd *ReturnDetail) Validate() error {
+func (rd *ReturnDetail) Validate(opts ...*ValidateOpts) error {
```

Comment:
In ACH we passed the ValidateOpts through a file so all records in a file have the same validation overrides.

Reply:
Right, ACH used above logic 
but imagecashletter is different with ACH
Actually we need to hit validation function of ReturnDetail and CheckDetail 
Validation function of file don't hit these validation functions
The validation functions will hit only when parsing file

I am not sure the reason yet. anyway we should use this parameter for supporting current logic

Reply:
cc @atonks2 thoughts?

Reply:
ACH adds an instance of `validateOpts` to several record types, but then still exposes a `ValidateWith` method on some of them to pass opts through. 

If we add a validateOpts field to each record that has options defined and wire it all the way through from file creation, that would probably cover the most code paths (trying to think through HTTP API vs library). That said, it's not clear to me from a quick review of ACH how the opts get wired through.

@adamdecaf Can you provide more context on how the validate opts work in ACH? How/when are they wired through when using ACH as an API vs a library.

Reply:
They're wired through from the `File` object so you can accumulate records and the opts are respected during validation. The HTTP api accepts them as a top-level object in JSON. 

The ACH library's `Reader` can also have `ValidateOpts` which are put onto each file that's read.